app/src/main/java/com/example/firstapplication/MainActivity.kt

```diff
@@ -7,28 +7,30 @@ import java.util.*
 
 class MainActivity : AppCompatActivity() {
     private lateinit var binding: ActivityMainBinding
+    private lateinit var usersViewModel: UserListViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        val userAdapter = UserAdapter(
-            mutableListOf(
-                User("user-id-1", "user-name-1", "user-1-email"),
-                User("user-id-2", "user-name-2", "user-2-email"),
-                User("user-id-3", "user-name-3", "user-3-email"),
-                User("user-id-4", "user-name-4", "user-4-email"),
-                User("user-id-5", "user-name-5", "user-5-email")
-            )
-        )
         val recyclerView = binding.recyclerView
-        recyclerView.adapter = userAdapter
 
+        usersViewModel = UserListViewModel()
```

Comment:
Nie tworzymy viewmodelu wewnątrz aktywności. ViewModel z założenia może "żyć" dłużej niż aktywność. Instancja Activity może być zniszczona i jej stan może być przywrócony w nowej instancji np. w przypadku zmiany konfiguracji ekranu. Założeniem ViewModelu było to, by pozwolić danym oraz połączeniom z serwerem czy bazą danych przetrwać gdy aktywność jest niszczona. Jeśli więc stworzyć jej instancję tutaj, to przy odtworzeniu aktywności stworzysz nowy obiekt viewModel, zamiast podłączyć się do istniejącego.

Więcej o tym co napisałem oraz o tym jak uzyskać instancję ViewModel znajdziesz tutaj: https://developer.android.com/topic/libraries/architecture/viewmodel
pierwsze dwa rozdziały

Reply:
tak mi się wydawało że nie zrozumiałam tej idei :) teraz już jasne :) dzięki 👍